### PR TITLE
[MNT] Update codeowners

### PR DESCRIPTION
**Pull request by TonyBagnall**
#### What does this implement/fix? Explain your changes.

Initially my intention was to remove inactive core devs who no longer want to be listed as code owners. However, I notice that a lot of code does not have code owners listed. I have put place holders for these. I'm not sure if this wanted though. I am not aiming for any controversy, I simply think it a good idea to update this and have a complete list of packages. For example, I wasnt sure if anyone should "own" tests or datatypes. Anyone feel free to update this PR, remove/add owners remove/add packages not "owned".  

Is @GuzalBulatova meant to be listed at the top of this file?

* @fkiraly @aiwalter

### CODE REVIEW

File: CODEOWNERS
Comment (fkiraly) [inline CODEOWNERS:28]: why this? Not your algorithm, @TonyBagnall

**Review by fkiraly — CHANGES_REQUESTED** [review]
@TonyBagnall, have the people that you are removing agreed to being removed?

Where you are adding yourself as a code owner, have the previous owners agreed?

**Comment by fkiraly** [comment]
I think either the original owners should open a PR, or give their explicit consent by other means, e.g., issue/PR comment, email to sktime CC, or similar.

From the aspect of it, it looks look like the primary beneficiary of this PR in terms of "ownership" is you yourself, so the consent should be via a channel independent of you.

**Comment by TonyBagnall** [comment]
> From the aspect of it, it looks look like the primary beneficiary of this PR in terms of "ownership" is you yourself, so the consent should be via a channel independent of you.

"beneficiary" is a strange term. I have added myself to clustering and distances. Signatures add was a typo. I am not doing this as some kind of power play. Any others you contest or disagree with, just post here, I do not in principle mind being removed as code owner. I am just trying to make this file up to date.

**Comment by TonyBagnall** [comment]
> @TonyBagnall, have the people that you are removing agreed to being removed?
> 
> Where you are adding yourself as a code owner, have the previous owners agreed?

where exactly do you mean? Signatures was a typo, I have added myself to clustering and distances.

> > Is @GuzalBulatova meant to be listed at the top of this file?
> 
> Yes, you should have seen that in the internal slack channel where we were discussing precisely this.

do you want to add her to the file then or would you rather I do it?

**Comment by fkiraly** [comment]
> "beneficiary" is a strange term. I have added myself to clustering and distances. Signatures add was a typo. I am not doing this as some kind of power play. Any others you contest or disagree with, just post here, I do not in principle mind being removed as code owner. I am just trying to make this file up to date.

There is no implication of intent here, Tony. You are adding yourself as an owner in a lot of places. and you need to ask people for consent.

**Comment by TonyBagnall** [comment]
@fkiraly "There is no implication of intent here, Tony. You are adding yourself as an owner in a lot of places. and you need to ask people for consent." This confused me, because I did not think this PR was about me at all, nor that I had added myself to anything that was already owned. On main, I am listed as owner here

sktime/classification/dictionary_based/ @patrickzib @MatthewMiddlehurst @TonyBagnall
sktime/classification/distance_based/ @jasonlines @goaster @TonyBagnall
sktime/classification/early_classification/ @MatthewMiddlehurst @TonyBagnall
sktime/classification/early_classification/_teaser.py @patrickzib @MatthewMiddlehurst @TonyBagnall
sktime/classification/feature_based/ @MatthewMiddlehurst @TonyBagnall
sktime/classification/feature_based/_signature_classifier.py @jambo6
sktime/classification/hybrid/ @MatthewMiddlehurst @TonyBagnall
sktime/classification/interval_based/ @MatthewMiddlehurst @TonyBagnall
sktime/classification/kernel_based/ @MatthewMiddlehurst @TonyBagnall
sktime/classification/shapelet_based/ @jasonlines @ABostrom @TonyBagnall
sktime/classification/shapelet_based/_stc.py @jasonlines @ABostrom @MatthewMiddlehurst @TonyBagnall
sktime/transformations/panel/shapelet_transform.py @MatthewMiddlehurst @TonyBagnall
sktime/transformations/panel/shapelets.py @jasonlines @ABostrom @TonyBagnall
sktime/transformations/panel/channel_selection.py @haskarb @a-pasos-ruiz @TonyBagnall

on this PR I have actually removed myself as owner from here 
sktime/classification/dictionary_based/ @patrickzib @MatthewMiddlehurst
sktime/classification/distance_based/
sktime/classification/early_classification/ @MatthewMiddlehurst
sktime/classification/early_classification/_teaser.py @patrickzib @MatthewMiddlehurst @TonyBagnall

I have added new ownerships here
sktime/clustering/ @chrisholder @TonyBagnall
sktime/distances/ @chrisholder @TonyBagnall

where previously there was no code owner listed. Is it these two you refer to? @chrisholder do you mind me being listed as an owner? I did by mistake list myself on signatures as you pointed out, which I have corrected. Sorry if I am missing something, I have tried to figure out what module you are referring to. Could you clarify who I need to ask consent from and for what module? thanks, 

One thing I had hoped @fkiraly is that you would claim ownership of some of the orphaned modules that I believe you wrote, such as sktime/alignment/, sktime/datatypes/, sktime/dists_kernels/, sktime/param_est/, sktime/pipeline, sktime/series_as_features, sktime/classification/distance_based/. It would be good to have owners of all modules imo. Also I wasnt sure whether to list tests/ as a module.

**Comment by fkiraly** [comment]
@TonyBagnall, I think some things aren't quite as you say.
Let's clarify.

> One thing I had hoped @fkiraly is that you would claim ownership of some of the orphaned modules that I believe you wrot

The top-level modules are not "owned".
Only estimators can be owned. See our governance document
https://www.sktime.org/en/latest/get_involved/governance.html#governance

In particular, the classification module, as a full module, does not have any owners (e.g., you), even if it has people who prefer to work only within that module (e.g., you).

The `sktime` community acts as owners/curators of any non-algorithm content, e.g., framework code, or algorithms that their authors prefer not to own.

**Comment by fkiraly** [comment]
I'm not going to produce a full list of this, but let me list examples for the different things you are doing:

* you remove other individuals as owners, e.g., `distance_based`. That requires consent and approval of anyone you remove
* you add other individuals as owners, e.g., `early_classification._teaser`. That requires consent and approval of anyone you add
* you add owners to top-level modules which are not "owned", e.g., `clustering`, `distances`, `early_classification`. In some instances the added owner is you yourself. This shouldn't be done at all (see above)
* you add yourself as an owner to someone else's algorithm, e.g., `_signature_classifier`. That requires consent and approval and consent of the current owner.

**Review by fkiraly — CHANGES_REQUESTED** [review]
See above.

**Comment by patrickzib** [comment]
> you add other individuals as owners, e.g., early_classification._teaser. That requires consent and approval of anyone you add, as well as the current owners (where applicable).

As I see it:  he removed himself from these packages but was listed before? So anyhow. Both is fine with me.

**Comment by jasonlines** [comment]
Just to confirm that @TonyBagnall did contact me directly about code ownership before making this PR and I consented to the change. I'm hoping to become more active in sktime again in general in the future but unfortunately I don't have the resources to contribute consistently at the moment. I'm already not listed as an _active_ core dev due to this so it makes sense to tidy up ownerships to make sure the files in question are properly maintained by active devs. 

> I think either the original owners should open a PR, or give their explicit consent by other means, e.g., issue/PR comment, email to sktime CC, or similar.

Thanks for checking @fkiraly and it may be best going forward to agree on a process for updating code ownership to avoid future confusion. 

For my two cents, as someone involved in it this time, I think @TonyBagnall's approach of opening a PR and proactively changing the ownerships is a sensible approach because code owners who stop engaging with sktime alltogether may not open a PR themself/comment on a PR to confirm. Perhaps it would be sensible in the future to open a PR like @TonyBagnall has, and explicitly give the code owners in question a reasonable amount of time to respond on the PR before it is merged (i.e. 2 weeks? more/less?). This would give current owners a chance to respond in cases where they wish to remain as an owner, but it would also avoid any issues with code owners not engaging at all. 

Anyway, just a suggestion - TL;DR @TonyBagnall contacted me about the proposed changed before changing my ownership and I was/am happy to support it

**Comment by fkiraly** [comment]
>it may be best going forward to agree on a process for updating code ownership to avoid future confusion.

There already is a (mini-)process here:
https://www.sktime.org/en/stable/get_involved/governance.html#algorithm-maintainers
It requires consent of the original maintainer.

> I think @TonyBagnall's approach of opening a PR and proactively changing the ownerships is a sensible approach because code owners who stop engaging with sktime alltogether may not open a PR themself/comment on a PR to confirm.

I disagree that this is agood process - algorithm maintainers may feel strongly about ownership, but look only every few months after their algorithm.
Putting an onus on them of checking every week or someone can take over your algorithm is a bad idea in my opinion, and also unfair.

Why don't just leave them forever owning it if they disengage, per current policy anyone can add a new estimator which could even be a full copy of the other estimator (they just need to cite the old estimator and its author properly).

**Pull request closed without merging**
